modules/optimization/src/optimizationreporters/OptimizationReporterBase.C

```diff
@@ -9,17 +9,70 @@
 
 #include "OptimizationReporterBase.h"
 
+namespace // unnamed namespace
+{
+void
+copyReporterIntoPetscVector(const std::vector<std::vector<Real> *> reporterVectors,
+                            libMesh::PetscVector<Number> & x)
+{
+  dof_id_type n = 0;
+  for (const auto & data : reporterVectors)
+    for (const auto & val : *data)
+      x.set(n++, val);
+
+  x.close();
+}
+
+void
+copyPetscVectorIntoReporter(const libMesh::PetscVector<Number> & x,
+                            std::vector<std::vector<Real> *> reporterVectors)
+{
+  dof_id_type n = 0;
+  for (auto & data : reporterVectors)
+    for (auto & val : *data)
+      val = x(n++);
+}
+}
+
```

Comment:
I'd prefer to see these as public static functions of `OptimizationReporterBase`